src/core/server/execution_context/execution_context_container.ts

```diff
@@ -57,7 +57,11 @@ export class ExecutionContextContainer implements IExecutionContextContainer {
   }
   toString(): string {
     const ctx = this.#context;
-    const contextStringified = ctx.type && ctx.id ? `kibana:${ctx.type}:${ctx.id}` : '';
+    const contextStringified =
+      ctx.type && ctx.id
+        ? // id may contain non-ASCII symbols
+          `kibana:${encodeURIComponent(ctx.type)}:${encodeURIComponent(ctx.id)}`
```

Comment:
a visualization can be created via SO API, there is no guarantee that it contains only latin characters. https://github.com/elastic/kibana/blob/7b19c748c1724dcf520d767470ec54f9164b210a/x-pack/test/functional/es_archives/dashboard/drilldowns/data.json#L19

I will ping Elasticsearch team that they need to decode `x-opaque-id` header before emitting to the logs (if not yet)

Reply:
@pgomulka confirmed `slowlogs` contains a decoded `x-opaque-id` value in `source.query` field:
```json
{
   "type":"index_search_slowlog",
   "timestamp":"2021-07-13T11:42:33,366+02:00",
   "level":"TRACE",
   "component":"i.s.s.query",
   "cluster.name":"elasticsearch",
   "node.name":"Przemyslaws-MacBook-Pro.local",
   "message":"[test_1][0]",
   "took":"403.8micros",
   "took_millis":"0",
   "total_hits":"0 hits",
   "types":"[]",
   "stats":"[]",
   "search_type":"QUERY_THEN_FETCH",
   "total_shards":"1",
   "source":"{\"query\":{\"match\":{\"field\":{\"query\":\"kibana:execution_context_app::Visualization☺漢字\",\"operator\":\"OR\",\"prefix_length\":0,\"max_expansions\":50,\"fuzzy_transpositions\":true,\"lenient\":false,\"zero_terms_query\":\"NONE\",\"auto_generate_synonyms_phrase_query\":true,\"boost\":1.0}}}}",
   "id":"Visualization%E2%98%BA%E6%BC%A2%E5%AD%97",
   "cluster.uuid":"Ria5FHVtSeeuALzqZQWlXg",
   "node.id":"3RKIxU6PRr6S_M4YJGhVFA"
}
```